src/mono/msbuild/android/build/AndroidBuild.targets

```diff
@@ -41,40 +41,36 @@
       <_MonoHeaderPath>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'include', 'mono-2.0'))</_MonoHeaderPath>
     </PropertyGroup>
 
-    <!-- Make sure marshal-ilgen is included in the components list. -->
-    <ItemGroup Condition="'$(RuntimeComponents)' != '*'">
-      <_RuntimeComponentList Include="$(RuntimeComponents)" />
-      <_RuntimeComponentList Include="marshal-ilgen" KeepDuplicates="false"/>
+    <MonoRuntimeComponentManifestReadTask Condition="'$(UseNativeAOTRuntime)' != 'true'" JsonFilePath="$(_MonoRuntimeComponentManifestJsonFilePath)">
```

Comment:
May be worth moving this to the `MonoSharedBuild.targets` if it is a low effort.

Reply:
Do you mean to introduce a new targets file: `MonoSharedBuild.targets` ? 
That would make sense, but there are more things that could be shared between `AndroidBuild.targets` and `AppleBuild.targets` so I would leave it to a separate PR (like AndroidSample to use `AndroidBuild.targets` etc)

Reply:
Yes. I agree, not scope of this effort.

Reply:
There's probably more to share between all our configurations.

Reply:
Just commenting. I don't think it's necessary to do it in this pr.